src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -265,9 +266,15 @@ public function lists($column, $key = null)
      * @param  string  $pageName
      * @param  int|null  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
+     *
+     * @throws InvalidArgumentException
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage <= 0) {
+            throw new InvalidArgumentException("Negative values can't be used to query results");
```

Comment:
I have checked the other PR and I didn't seem to see/notice the exception message stated there. Please, can you just mention the exact message here?